src/Host/Client/Impl/Session/RSession.cs

```diff
@@ -748,81 +754,90 @@ private class BrokerTransaction : IRSessionSwitchBrokerTransaction, IRSessionRec
             }
 
             public async Task CompleteSwitchingBrokerAsync(CancellationToken cancellationToken) {
+                if (_session._disposeToken.IsDisposed) {
+                    return;
+                }
+
                 if (_lockToken == null) {
                     throw new InvalidOperationException($"{nameof(AcquireLockAsync)} must be called before {nameof(CompleteSwitchingBrokerAsync)}");
                 }
 
-                _session._disposeToken.ThrowIfDisposed();
-
-
                 if (_session._startupInfo == null) {
```

Comment:
Done

---
In reply to: [83917898](https://github.com/Microsoft/RTVS/pull/2529#discussion_r83917898) [](ancestors = 83917898)